test-dts/appUse.test-d.ts

```diff
@@ -0,0 +1,71 @@
+import { createApp } from './index'
+
+type App = ReturnType<typeof createApp>
+
+type PluginAOptionType = {
+  option1?: string
+  option2: number
+  option3: boolean
+}
+
+const PluginA = {
+  install(app: App, ...options: PluginAOptionType[]) {
+    options[0].option1
+    options[0].option2
+    options[0].option3
+  }
+}
+
+const PluginB = {
+  install(app: App) {}
+}
+
+const PluginC = (app: App, ...options: string[]) => {}
+
+createApp({})
+  .use(PluginA)
+  // @ts-expect-error option2 and option3 (required) missing
+  .use(PluginA, {})
+  // @ts-expect-error type mismatch
+  .use(PluginA, true)
+  // @ts-expect-error type mismatch
+  .use(PluginA, undefined)
+  // @ts-expect-error type mismatch
+  .use(PluginA, null)
+  // @ts-expect-error type mismatch
+  .use(PluginA, 'foo')
+  // @ts-expect-error type mismatch
+  .use(PluginA, 1)
+  .use(PluginA, { option2: 1, option3: true })
+  .use(PluginA, { option1: 'foo', option2: 1, option3: true })
+  .use(PluginA, { option1: 'foo', option2: 1, option3: true }, { option2: 1, option3: true })
```

Comment:
I see. So I don’t think this is good. I will give a proper review for the types and an alternative when I can but it shouldn’t be just `Option[]` because it usually allows a specific amount of arguments (options) to be passed to a plugin. Similar to emit()